app/controllers/posts_controller.rb

```diff
@@ -9,4 +9,32 @@ def show
     @likes = Post.find(params[:id]).likes
     @comments = Post.find(params[:id]).comments
   end
+
+  # GET /posts/new
+  def new
+    @post = Post.new
+  end
+
+  # POST /posts or /posts.json
+  def create
+    @post = Post.new(post_params.merge(comments_counter: 0, likes_counter: 0, author_id: current_user.id))
+    respond_to do |format|
+      if @post.save
+        format.html do
+          redirect_to "/users/#{@post.user.id}", notice: 'Post was successfully created.'
```

Comment:
- [ ] This should be `@post.current_user.id` not `@post.user.id`

Reply:
The @post.user.id is used to redirect to the posts page according to the route.